torch/csrc/onnx/init.cpp

```diff
@@ -39,7 +39,7 @@ void initONNXBindings(PyObject* module) {
 
   onnx.attr("PRODUCER_VERSION") = py::str(TORCH_VERSION);
 
-#ifdef PYTORCH_ONNX_CAFFE2_BUNDLE
+#ifdef BUILD_CAFFE2
   onnx.attr("PYTORCH_ONNX_CAFFE2_BUNDLE") = true;
```

Comment:
It seems like the CMake files have no reference to `PYTORCH_ONNX_CAFFE2_BUNDLE`, so is it even possible for this to be true?

If it's always false, then I think we should:
* Change the default of `operator_export_type` to always be `OperatorExportTypes.ONNX`. This just simplifies the code and docs while keeping current behavior.
* Change the name of this attr to be something private and more specific like `_CAFFE2_ATEN_FALLBACK`. This hides an implementation detail and makes the name more understandable. This would be a backwards-compatibility breaking change since we're removing a public attribute from the module, so if you do this please add a note to the PR like the one here: https://github.com/pytorch/pytorch/pull/66617

Reply:
Yup, `PYTORCH_ONNX_CAFFE2_BUNDLE` is not defined anywhere - just read in this single file. It really seems dead code unless there is a hacky way to set it (such as manually changing build/CMakeCache.txt before a rebuild?). In my experiments, it is always False for supported scenarios (with and without caffe2).

Relying on `BUILD_CAFFE2` feels less intrusive as it keeps an official (public) path to Caffe2 scenarios

Reply:
* Are suggesting `ONNX` as default even when Caffe2 is built with Pytorch? As it seems, Caffe2 is not usable along PyTorch without `ONNX_ATEN_FALLBACK` as operator_export_type

* I am ok with changing `PYTORCH_ONNX_CAFFE2_BUNDLE` to `_CAFFE2_ATEN_FALLBACK`

Reply:
> Are suggesting ONNX as default even when Caffe2 is built with Pytorch?

Yes. Users can explicitly request fallback if they want, which is already the case today since we determined that `PYTORCH_ONNX_CAFFE2_BUNDLE` is always `False`.

Reply:
done